File: Data Structures & Algorithms/validate-parentheses/submission-5.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        count={"}":"{", "]":"[", ")":"("}

        for i in s:
            if i in count:
                if stack and stack[-1]==count[i]:
                    stack.pop()
                else: return False
            else: stack.append(i)
        return True if not stack else False

File: Data Structures & Algorithms/minimum-stack/submission-5.cpp
class MinStack {
public:
        stack<int>ms;
        stack<int>s;
    MinStack() {
    }
    
    void push(int val) {
        s.push(val);
        ms.push(min(val,ms.empty()?val:ms.top()));
    }
    
    void pop() {
        s.pop();
        ms.pop();
    }
    
    int top() {
        return s.top();
    }
    
    int getMin() {
        return ms.top();
    }
};
